src/cart/crt_swim.c

```diff
@@ -326,6 +347,30 @@ crt_swim_update_delays(struct crt_swim_membs *csm, uint64_t hlc,
 	return snd_delay;
 }
 
+/*
+ * If id belongs to the primary group, this function returns 0; otherwise, this
+ * function returns the group version. Note that if the group version is 0,
+ * that is, the primary group has not been initialized yet, this function
+ * always returns 0.
+ */
+static uint32_t
+crt_swim_lookup_id(swim_id_t id)
+{
+	struct crt_grp_priv	*grp_priv = crt_gdata.cg_grp->gg_primary_grp;
+	d_rank_list_t		*membs;
+	uint32_t		 grp_ver;
+
+	D_RWLOCK_RDLOCK(&grp_priv->gp_rwlock);
+	grp_ver = grp_priv->gp_membs_ver;
+	membs = grp_priv_get_membs(grp_priv);
+	if (membs) {
+		if (d_rank_in_rank_list(membs, id))
```

Comment:
We need TODO here to note that this is a costly operation today

Reply:
Sure. Will add one.